packages/framework-types/src/provider.ts

```diff
@@ -35,16 +37,22 @@ export interface ProviderEventsLibrary {
     entityID: UUID,
     since?: string
   ): Promise<Array<EventEnvelope>>
-  latestEntitySnapshot(config: BoosterConfig, entityTypeName: string, entityID: UUID): Promise<EventEnvelope | null>
+  latestEntitySnapshot(
+    config: BoosterConfig,
+    entityTypeName: string,
+    entityID: UUID
+  ): Promise<EntitySnapshotEnvelope | undefined>
   search(config: BoosterConfig, parameters: EventSearchParameters): Promise<Array<EventSearchResponse>>
   searchEntitiesIDs(
     config: BoosterConfig,
     limit: number,
     afterCursor: Record<string, string> | undefined,
     entityTypeName: string
   ): Promise<PaginatedEntitiesIdsResult>
-  /** Streams an event to the corresponding event handler */
-  store(eventEnvelopes: Array<EventEnvelope>, config: BoosterConfig): Promise<void>
+  /** Stores a list of events in the events store */
+  store(eventEnvelopes: Array<NonPersistedEventEnvelope>, config: BoosterConfig): Promise<void>
+  /** Stores an entity snapshot in the event store */
+  storeSnapshot(snapshotEnvelope: EntitySnapshotEnvelope, config: BoosterConfig): Promise<void>
```

Comment:
This PR introduces a dedicated method in the `ProviderLibrary` interface to store snapshots.